<a href="https://colab.research.google.com/github/pishanjia/ADS_REV_code/blob/main/Capstone.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np
# importing needed libraries for cleaning,analysis and visualization





In [21]:
#Loading and inspecting datasets

Basics=pd.read_csv("/content/title.basics1.csv")
Ratings=pd.read_csv("/content/title.ratings.csv")
Bom=pd.read_csv("/content/bom.movie_gross.csv")






In [24]:
Basics.head(10)


tconst                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   
5  tt0111414                      A Thin Life                 A Thin Life   
6  tt0112502                          Bigfoot                     Bigfoot   
7  tt0137204                  Joe Finds Grace             Joe Finds Grace   
8  tt0139613                      O Silêncio                 O Silêncio   
9  tt0144449            Nema aviona za Zagreb       Nema aviona za Zagreb   

   start_year  runtime_minutes                      genres  
0        2013            175.0          Action,Crime,Drama  
1        2019            114.0             Biography,Drama  
2        2018            122.0                       Drama  
3        2018              NaN                Comedy,Drama  
4        2017             80.0        Comedy,Drama,Fantasy  
5        2018             75.0                      Comedy  
6        2017              NaN             Horror,Thriller  
7        2017             83.0  Adventure,Animation,Comedy  
8        2012              NaN         Documentary,History  
9        2012             82.0                   Biography

In [25]:
Ratings.head(10)

tconst  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21
5   tt1069246            6.2       326
6   tt1094666            7.0      1613
7   tt1130982            6.4       571
8   tt1156528            7.2       265
9   tt1161457            4.2       148

In [22]:
Bom.head(10)

tconst                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   
5  tt0111414                      A Thin Life                 A Thin Life   
6  tt0112502                          Bigfoot                     Bigfoot   
7  tt0137204                  Joe Finds Grace             Joe Finds Grace   
8  tt0139613                      O Silêncio                 O Silêncio   
9  tt0144449            Nema aviona za Zagreb       Nema aviona za Zagreb   

   start_year  runtime_minutes                      genres  
0        2013            175.0          Action,Crime,Drama  
1        2019            114.0             Biography,Drama  
2        2018            122.0                       Drama  
3        2018              NaN                Comedy,Drama  
4        2017             80.0        Comedy,Drama,Fantasy  
5        2018             75.0                      Comedy  
6        2017              NaN             Horror,Thriller  
7        2017             83.0  Adventure,Animation,Comedy  
8        2012              NaN         Documentary,History  
9        2012             82.0                   Biography

Statistical summary

In [28]:
Basics.describe()


domestic_gross         year
count    3.359000e+03  3387.000000
mean     2.874585e+07  2013.958075
std      6.698250e+07     2.478141
min      1.000000e+02  2010.000000
25%      1.200000e+05  2012.000000
50%      1.400000e+06  2014.000000
75%      2.790000e+07  2016.000000
max      9.367000e+08  2018.000000

In [29]:
Ratings.describe()

averagerating      numvotes
count   73856.000000  7.385600e+04
mean        6.332729  3.523662e+03
std         1.474978  3.029402e+04
min         1.000000  5.000000e+00
25%         5.500000  1.400000e+01
50%         6.500000  4.900000e+01
75%         7.400000  2.820000e+02
max        10.000000  1.841066e+06

In [30]:
Bom.describe()

domestic_gross         year
count    3.359000e+03  3387.000000
mean     2.874585e+07  2013.958075
std      6.698250e+07     2.478141
min      1.000000e+02  2010.000000
25%      1.200000e+05  2012.000000
50%      1.400000e+06  2014.000000
75%      2.790000e+07  2016.000000
max      9.367000e+08  2018.000000

In [31]:
Basics.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   tconst           146144 non-null  object 
 1   primary_title    146143 non-null  object 
 2   original_title   146122 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


In [32]:
Ratings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73856 entries, 0 to 73855
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   tconst         73856 non-null  object 
 1   averagerating  73856 non-null  float64
 2   numvotes       73856 non-null  int64  
dtypes: float64(1), int64(1), object(1)
memory usage: 1.7+ MB


In [33]:
Bom.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


Checking Missing Values

In [35]:
Basics.isnull().sum()

tconst                 0
primary_title          1
original_title        22
start_year             0
runtime_minutes    31739
genres              5408
dtype: int64

In the above dataset we have
1 missing value in primary title
22 missing value in the original title
0 missing vlaues for start_year
31739 missing vlaues for runtime minutes
5408 missing values under genre

In [38]:
Ratings.isnull().sum()

tconst           0
averagerating    0
numvotes         0
dtype: int64

This dataset has no missing values

In [39]:
Bom.isnull().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

This dataset has the following missing values
5 missing values for the studio
28 domestic gross
1350 foreign gross



Handling missing values

Cleaning title Basics Data

In [42]:
runtime_missing = Basics['runtime_minutes'].median()
Basics["runtime_minutes"].fillna(runtime_missing)


0         175.0
1         114.0
2         122.0
3          87.0
4          80.0
          ...  
146139    123.0
146140     87.0
146141     87.0
146142    116.0
146143     87.0
Name: runtime_minutes, Length: 146144, dtype: float64

In [46]:
Basics['primary_title'].fillna("unknown")

0                                            Sunghursh
1                      One Day Before the Rainy Season
2                           The Other Side of the Wind
3                                      Sabse Bada Sukh
4                             The Wandering Soap Opera
                              ...                     
146139                             Kuambil Lagi Hatiku
146140    Rodolpho Teóphilo - O Legado de um Pioneiro
146141                                 Dankyavar Danka
146142                                          6 Gunn
146143                Chico Albuquerque - Revelações
Name: primary_title, Length: 146144, dtype: object

In [43]:
Basics['genres'].fillna("unknown")

0           Action,Crime,Drama
1              Biography,Drama
2                        Drama
3                 Comedy,Drama
4         Comedy,Drama,Fantasy
                  ...         
146139                   Drama
146140             Documentary
146141                  Comedy
146142                 unknown
146143             Documentary
Name: genres, Length: 146144, dtype: object

In [44]:
Basics_cleaned=Basics.dropna(subset=["original_title"])

In [48]:
Basics_cleaned.info()

<class 'pandas.core.frame.DataFrame'>
Index: 146122 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   tconst           146122 non-null  object 
 1   primary_title    146122 non-null  object 
 2   original_title   146122 non-null  object 
 3   start_year       146122 non-null  int64  
 4   runtime_minutes  146122 non-null  float64
 5   genres           140733 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 7.8+ MB


Cleaning BOM Dataset

In [49]:
Bom['studio'].fillna("unknown")

0               BV
1               BV
2               WB
3               WB
4             P/DW
           ...    
3382         Magn.
3383            FM
3384          Sony
3385    Synergetic
3386         Grav.
Name: studio, Length: 3387, dtype: object

In [50]:
Bom['foreign_gross'] = pd.to_numeric(Bom['foreign_gross'], errors='coerce')

In [53]:
Bom_movie_cleaned = Bom.dropna(subset=['domestic_gross'])